### PR TITLE
Pico support for adxl355 and pico platform changes

### PR DESCRIPTION
**Pull request by rbolboac**
- Initialize uart with stdio API to be able to use stdio related functions (e.g. printf).
- Remove wrap function for ldivmod to avoid linker errors when no_os_util is used.
- Rework spi driver for pico platform: use CS as GPIO (if used as SPI function the CS is deasserted by
default after every 8 bits), use pico_spi_transfer in pico_spi_write_and_read to remove duplicated code, use only chip_select for the slave_id creation
- Add support for pico platform for eval-adxl355-pmdz for dummy example, iio example and iio example with gpio trigger.

### CODE REVIEW

**Comment by rbolboac** [comment]
- removed extra line

File: projects/eval-adxl355-pmdz/src/platform/pico/parameters.c [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+};
+
+struct pico_uart_init_param adxl355_uart_extra_ip = {
+	.uart_tx_pin = UART_TX_PIN,
+	.uart_rx_pin = UART_RX_PIN,
+};
+
+#ifdef IIO_TRIGGER_EXAMPLE
+
+#endif
```
Comment (amiclaus) [inline projects/eval-adxl355-pmdz/src/platform/pico/parameters.c:62]: nit: new line at the end of the file required.

File: projects/eval-adxl355-pmdz/src/platform/pico/parameters.c [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+	.spi_sck_pin = SPI0_SCK_GP18,
+	.spi_cs_pin = SPI0_CS_GP17
+};
+
+struct pico_uart_init_param adxl355_uart_extra_ip = {
+	.uart_tx_pin = UART_TX_PIN,
+	.uart_rx_pin = UART_RX_PIN,
+};
+
+#ifdef IIO_TRIGGER_EXAMPLE
```
Comment (amiclaus) [inline projects/eval-adxl355-pmdz/src/platform/pico/parameters.c:60]: what's the purpose of this `ifdef`? i'd add it only when implementation is added.

**Review by amiclaus — APPROVED** [review]

File: projects/eval-adxl355-pmdz/src/platform/pico/parameters.h [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+#include "pico_timer.h"
+
+/******************************************************************************/
+/********************** Macros and Constants Definitions **********************/
+/******************************************************************************/
+
+#define UART_DEVICE_ID      0
+#define UART_BAUDRATE  115200
+#define UART_IRQ_ID        20
+#define UART_EXTRA       &adxl355_uart_extra_ip
```
Comment (CiprianRegus) [inline projects/eval-adxl355-pmdz/src/platform/pico/parameters.h:62]: These have different number of tabs.

**Review by CiprianRegus — APPROVED** [review]
Looks good!

**Comment by rbolboac** [comment]
- changed indentation

**Pull request merged**